src/containers/Dashboard.vue

```diff
@@ -181,56 +163,60 @@ export default Vue.extend({
+    const loadSemesterPromise = semesterDoc.get().then(snapshot => {
+      const data = snapshot.data();
+      if (data != null) {
+        this.semesters = data.semesters;
+      } else {
+        const newSemeter: FirestoreSemester = {
+          type: getCurrentSeason(),
+          year: getCurrentYear(),
+          courses: [],
+        };
+        this.semesters = [newSemeter];
+        semesterDoc.set({ semesters: [newSemeter] });
+      }
+    });
+    const loadOtherDataPromise = new Promise<void>(resolve => {
+      listenerUnsubscriber = initializeFirestoreListeners(() => resolve());
+    });
+    Promise.all([loadSemesterPromise, loadOtherDataPromise]).then(() => {
+      if (this.onboardingData.college !== '') {
+        this.loaded = true;
+        this.recomputeRequirements();
+      } else {
+        this.startOnboarding();
+      }
+    });
+    subscribeRequirementDependencyChange(() => {
       // Avoid recomputing requirement before user data is initialized.
       if (this.onboardingData.college !== '') this.recomputeRequirements();
     });
   },
   destroyed() {
     window.removeEventListener('resize', this.resizeEventHandler);
+    listenerUnsubscriber();
   },
   methods: {
-    getUserEmail(): string {
-      return store.state.currentFirebaseUser.email;
-    },
-
-    getInformationFromUser() {
-      Promise.all([
-        usernameCollection.doc(this.getUserEmail()).get(),
-        semestersCollection.doc(this.getUserEmail()).get(),
-        onboardingDataCollection.doc(this.getUserEmail()).get(),
-      ]).then(([usernameDoc, semesterDoc, onboardingDataDoc]) => {
-        const usernameData = usernameDoc.data();
-        const semestersData = semesterDoc.data();
-        const onboardingData = onboardingDataDoc.data();
-        if (usernameData != null && onboardingData != null) this.userName = usernameData;
-        if (onboardingData != null) this.onboardingData = createAppOnboardingData(onboardingData);
-        if (semestersData != null) {
-          this.semesters = semestersData.semesters;
-        } else {
-          const newSemesterData = [
-            { type: getCurrentSeason(), year: getCurrentYear(), courses: [] },
-          ];
-          this.semesters = newSemesterData;
-          semestersCollection.doc(this.getUserEmail()).set({ semesters: newSemesterData });
-        }
-        if (usernameData != null && semestersData != null && onboardingData != null) {
-          this.loaded = true;
-          this.recomputeRequirements();
-        } else {
-          this.startOnboarding();
-        }
-      });
-    },
-
     editSemesters(newSemesters: readonly FirestoreSemester[]) {
+      console.log(
+        newSemesters.map(it => ({ a: `${it.type}${it.year}`, c: it.courses.map(c => c.code) }))
+      );
       this.semesters = newSemesters;
-      this.recomputeRequirements();
+      // this.recomputeRequirements();
```

Comment:
Can we remove this comment?

Reply:
ahh i'm doing some debugging, and it turns out this is the reason why the requirement bar doesn't update